#28 - implemented CipherFactory for PlainTextCipher

Zuhlek opened the pull request:
template implementation for the PlainTextCipher, future ciphers must be added to static section in CipherFactory.class

However, no use in current version, may get relevant later versions

Commit:
Add exemplary plain text implementation for a Cipher

Commit:
#27 - Implement first version of KeyGenerator class including test cases

Commit:
#27 - linked the parameters with the Key class, refactored related methods, created additional tests

Commit:
Merge remote-tracking branch 'origin/dev-branch' into 27-KeyGenerator-implementation

# Conflicts:
#	src/main/java/ch/psc/domain/cipher/Cipher.java
#	src/main/java/ch/psc/domain/cipher/Key.java
#	src/main/java/ch/psc/domain/cipher/PlainTextCipher.java
#	src/main/java/ch/psc/domain/file/PscFile.java

Commit:
#27 - Implemented JSON Adapter for Key.class, refactored various constructors/parameters for merging purposes, keyChain is now stored within the user.class, updated all relevant tests accordingly

Commit:
#28 - implemented CipherFactory for PlainTextCipher, no use in current version, will get relevant later...

SandroGuerotto reviewed (changes requested):
CipherFactory: sollte in createCipher ein Enum von allen Verfügbaren cipher-methoden kriegen.
Via switch-case kann dann entsprechend die Klasse zurückgegeben werden.
Eine Map ist unnötig und macht es komplizierter.

Zudem fehlt ein Enum, welches aussagt, ob der Algo verfügbar ist.

Commit:
Merge branch '27-KeyGenerator-implementation' into feature/46-GUI_AuswahlVerschlüsselungsmethode

# Conflicts:
#	src/main/java/ch/psc/domain/cipher/PlainTextCipher.java
#	src/main/java/ch/psc/domain/cipher/PscCipher.java
#	src/test/java/ch/psc/domain/cipher/PlainTextCipherTest.java

Commit:
#46 - Screen für encryption Auswahl implementiert (ChooseEncryption), Enum für die möglichen Encryptions erstellt für besseres handling, user mit neuer Eigenschaft (keyChain) vom Typ Map<String, Key> ausgestattet.

Commit:
More sophisticated CipherFactory using reflection

Single point of contact is Enum CipherAlgorithms. It needs to be kept up to date for both the CipherFactory and the GUI

SandroGuerotto reviewed (approved):
OK, danke für die Anpassungen

Commit:
Merge remote-tracking branch 'origin/dev-branch' into feature/28-CiphterFactoryImplementation

# Conflicts:
#	src/main/java/ch/psc/datasource/JSONWriterReader.java
#	src/main/java/ch/psc/domain/cipher/PlainTextCipher.java
#	src/main/java/ch/psc/domain/cipher/PscCipher.java
#	src/test/java/JSONWriterReaderTest.java
